### ISSUE TITLE
Call `compileModule` for `.svelte.js` files in Svelte 5

### ISSUE THREAD

**Issue by mcous** [svelteness/svelte-jester#282]
Howdy! I'm working on full Svelte 5 support over in [@testing-library/svelte](https://github.com/testing-library/svelte-testing-library/pull/375) and I discovered that our suite [fails](https://github.com/testing-library/svelte-testing-library/actions/runs/9103141513/job/25024344166?pr=375) with Svelte 5 + Jest + `svelte-jester`.

In Svelte 5, we rely on the `$state` rune to be able to re-render the component under test. This logic is placed in a `.svelte.js` file. However, if I try to send `.svelte.js` files through `svelte-jester`...

```ts
transform: {
  '^.+\\.svelte(?:\\.js)?$': ['svelte-jester'],
},
```

...Svelte becomes quite unhappy.

I gave `vite-plugin-svelte` a quick read, and discovered a new (to me) [part of the transformation pipeline](https://github.com/sveltejs/vite-plugin-svelte/blob/60fa8b4b43bce778268be67affcb929a0c4e4a2d/packages/vite-plugin-svelte/src/index.js#L189-L213). It looks like there's a new compiler function - `compileModule` - used to compile `.svelte.js` files.

**TL;DR: svelte-jester should probably know how to call `compileModule` for Svelte 5 support.**

I've got a very rough proof-of-concept locally that adds enough support to get our test suite passing. Is this something you'd accept a PR for?

**Comment by sebastianrothe** [comment]
> Hi @sebastianrothe, I hope you had a good weekend! Do you think you'll have time this week for these Svelte 5 PRs? I'm getting eager to ship proper Svelte 5 support in `@testing-library/svelte`, and I would prefer not to break Jest support while doing so

Yes, I am finally on it this evening.

**Comment by sebastianrothe** [comment]
I need to figure out tomorrow, how to release a *next*-Release for you to test.

**Comment by mcous** [comment]
Great! Let me know if I can help. For the time being, I'll try out a local build + tarball and report back

**Comment by sebastianrothe** [comment]
I think you can also use the git repo and main branch as a version number.

**Comment by mcous** [comment]
> I think you can also use the git repo and main branch as a version number.

Unfortunately, due to the build step, this doesn't work. The GitHub repo has no `dist` directory, where the main export is output by rollup

> I'll try out a local build + tarball and report back

Everything is checking out locally on my end! In Svelte 5, I am now able to specify...

```ts
{
  transform: {
    '^.+\\.svelte(?:\\.js)?$': 'svelte-jester',
  },
}
```

...in Svelte 5 and the v5 syntax (runes) works outside Svelte components as expected

**Comment by sebastianrothe** [comment]
Great, I will create a release with version number 5.

**Comment by sebastianrothe** [comment]
Published as `5.0.0` https://www.npmjs.com/package/svelte-jester 🚀
